### PR TITLE
Requested Sneak Changes & More

### PR DESCRIPTION
**Pull request by Vizoplez**
# THIS SHOULD BE TESTMERGED FIRST FOR BALANCE TESTING AND BUG TESTING. WHILE LOCALLY EVERYTHING IS WORKING FINE AND AS INTENDED; THIS WILL LIKELY NOT BE THE CASE ON THE SERVER.

## About The Pull Request
-Fixes Sneaking being able to be broken via pointing at the tile people were snuck on
-Fixes admins being unable to see invisible people
-No longer uses Alpha = 0
-Makes attacking people, being hit, and being pushed de-sneak you for 6 seconds.
-Spells now automatically unstealth you for **25 SECONDS**. Mage cope incoming!

NEW FEATURES:
PERCEPTION now PASSIVELY scales vs sneak; while you wont be able to hear their movements, the higher your PER, the more likely you are to just see _through their stealth._

SNEAK now scales how INVISIBLE you are; have a high enough sneak level? Even with 20 PER; you're not being found! Keep in mind, this doesn't make you immune to being de-snuck from attacking/being attacked/being bumped into; and more importantly... Light and people _looking closely_ may still be able to find you...
With 13 PER and 3 SNEAK (Journeyman); I am at invisibility 28; but can see entities at invisibility 27. This is subject to change (you can passively see sneaking people, though its difficult as they're partially transparent!)

NOTE: The transparency thing is _ONLY_ for people that can see you with their PERCEPTION - everyone else flat-out can't see you!

image

image -Still in sneak very close to the light!

image - Not a high enough level to sneak during the day!

image

Your SNEAK skill will also now scale the amount of light you can take before de-sneaking!

At LEGENDARY you can walk around sneaked beneath trees, and in slightly dark areas during the day - as long as there isn't any additional lights! Even a kneestinger is enough to break your cover. (0.652 lums required, average in day is 0.58 without nearby lights)
image

At MASTER you can walk openly at dusk/dawn - similarly to EXPERT (0.56 at MASTER, 0.468 at EXPERT)

at EXPERT and below; you get a MASSIVE reduction to the radius of lights being able to de-sneak you!

image
image

## Why It's Good For The Game
WIP but works locally. I need to rip out and re-code a lot of this as its er. Pretty disgustingly bad code; but this should majorly improve sneaking gameplay; especially as now you can't be randomly beamed by some guy trying to point at you.

Sneaking, by itself; was very niche, buggy - and felt unfun to play against/counterplay/play with; lights would randomly clip and de-sneak you, rogues would stay in stealth to frag vampyres (and there was really nothing they could do); and in general, stealth was far more of a _combat utility_ than merely a utility.

This buffs stealth for everyone _slightly_; but classes that use stealth are going to be - f a r - more powerful when sneaking now; at the stipulation of massively nerfing stealth and sneak attacks - along with othersuch things!

Also fixes admins unable to see people using stealth.

### CODE REVIEW

File: code/modules/mob/mob_movement.dm [hunk 1]
```diff
@@ -601,16 +604,21 @@
 		used_time = max(used_time - (mind.get_skill_level(/datum/skill/misc/sneaking) * 8), 0)
 
 	if(rogue_sneaking) //If sneaking, check if they should be revealed
-		if((stat > SOFT_CRIT) || IsSleeping() || (world.time < mob_timers[MT_FOUNDSNEAK] + 30 SECONDS) || !T || reset || (m_intent != MOVE_INTENT_SNEAK) || light_amount >= rogue_sneaking_light_threshhold)
+		if((stat > SOFT_CRIT) || IsSleeping() || (world.time < mob_timers[MT_FOUNDSNEAK] + 30 SECONDS) || !T || reset || (m_intent != MOVE_INTENT_SNEAK) || light_amount >= rogue_sneaking_light_threshhold || (world.time < mob_timers[MT_SNEAKATTACK] + 6 SECONDS) || (world.time < mob_timers[MT_SPELLSNEAK] + 25 SECONDS)) //SEVERELY nerfs sneak attacking.
```
Comment (Cyprex) [inline code/modules/mob/mob_movement.dm:607]: This needs to be refactored, this is *way* too long, ain't nobody gonna read all those conditions in one line. The !T check seems unnecessary from a cursory glance, but this should be taken apart and just rewritten, probably.

**Comment by Simply-Sydney** [comment]
Thank you for this, lack of admin sight on sneaking people was terrible

**Comment by HissyNays** [comment]
This seems like a flat out heavy nerf to sneaking.
The subtle transparency will not hide you and people that like to adjust their brightness for "better visibility" will certainly take advantage of this, now being able to see you in active stealth.
Why 25 seconds for mages?  This already seems like an anger-inspired change but that one line about mages really says yes, it's an angry-nerf-pr.

There's so much wrong with this that it would take me a really long message to explain why about each detail separately.

**Comment by HissyNays** [comment]
> > This seems like a flat out heavy nerf to sneaking. The subtle transparency will not hide you and people that like to adjust their brightness for "better visibility" will certainly take advantage of this, now being able to see you in active stealth. Why 25 seconds for mages? This already seems like an anger-inspired change but that one line about mages really says yes, it's an angry-nerf-pr.
> > There's so much wrong with this that it would take me a really long message to explain why about each detail separately.
> 
> Nobody noticed it and/or mentioned it ingame, so honestly I dont even know.
> 
> I was going to keep this open for a few more days to see if it needed balance adjustmetns but, yeah.

If people don't complain about it, it's because majority don't sneak. That isn't a reason to nerf a mechanic so heavily it's basicly removed.

If you wanted to do it reasonably then letting it make you be fully invisible at all skill levels is a must.
With skill levels changing movement speed while in sneak, how fast you turn invisible, how much darkness is required and how long you stay invisible after coming out of dark or turning off sneak. Then adding de-stealthed state which makes you somewhat transparent for 2 seconds when attacking or being attacked or using magic. 
It could also give you a de-stealthed state when you are moving in sneak with novice or no sneaking skill.

There we go, thats it, stealth is nerfed but in a reasonable way that keeps it useful but less potent in fights.

I have also heard that this change as whole is happening because some one got stabbed by a sneaking guy and no one could do anything apparently.
There is so much you can do to avoid that specificly and I've been in simmiliar situation and handled it well.

**Pull request merged**